workflows/methylation/methylation-preprocess.wdl

```diff
@@ -0,0 +1,64 @@
+version 1.1
+
+task process_raw_idats {
+    meta {
+        description: "Process raw IDAT files from the Illumina methylation array for a single sample"
+        outputs: {
+            beta_swan_norm_unfiltered: "Normalized beta values for all probes",
+            beta_swan_norm_unfiltered_genomic: "Normalized beta values for all probes that map to the genome",
+            annotation: "Annotation table for probes. Contains genomic coordinates and sequence and other information about the probes.",
+            beta_unnorm: "Non-normalized beta values",
+            cn_values: "Copy number values",
+            m_values: "M values",
+            probe_names: "Probe names found on the array",
+            sample_names: "Sample name found on the array",
+        }
+    }
+
+    parameter_meta {
+        idats: "Array of raw IDAT files from the Illumina methlyation array"
+        seed: "Random number generator seed for reproducibility. If processing samples independently, this should remain fixed for all runs that will be compared as a cohort."
+        disk_size_gb: "Disk size in GB"
+    }
+
+    input {
+        Pair[File, File] idats
+        Int seed = 1
+        Int disk_size_gb = 10
+    }
+
+    String idat_base = basename(idats.left)
+    String out_base = sub(idat_base, "(_Grn|_Red).idat", "")
+
+    command <<<
+        set -euo pipefail
+
+        ln -s ~{idats.left} ~{idats.right} .
```

Comment:
New conversation, same line of code so unresolving.

This will break if the `idat_base` isn't used by both files. We're already symlinking here, might as well go all the way and give the links the appropriate name.

Reply:
The challenge there is that `left` and `right` aren't necessarily the green or red channels. A user could swap them around. We could look for that suffix too, but if we're assuming that it is there, then I think it's fair to assume the first part is correct also. These files are generated by the array software, so unless someone mangles them, they should be reliable.

Reply:
Ah. So it's a bit more complicated than I realized. In that case, if we're going to make this assumption it should be documented in the `meta` somewhere visible. Preferably at both the task level and the workflow(s) level. But that has the problem of duplicating code...

Maybe a `warning` with the details at the task level, and then a `warning: "See task 'foo' for assumptions made"` at the workflow level?